CoreVisualSchema/test/unit/src/au/gov/asd/tac/constellation/graph/schema/visual/attribute/objects/ConnectionModeNGTest.java

```diff
@@ -0,0 +1,352 @@
+/*
+ * Copyright 2010-2022 Australian Signals Directorate
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package au.gov.asd.tac.constellation.graph.schema.visual.attribute.objects;
+
+import au.gov.asd.tac.constellation.graph.GraphReadMethods;
+import static org.mockito.ArgumentMatchers.anyInt;
+import org.mockito.Mockito;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.when;
+import static org.testng.Assert.assertEquals;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+/**
+ *
+ * @author serpens24
+ */
+public class ConnectionModeNGTest {
+
+    // Define mocks
+    GraphReadMethods mockGraphReadMethods;
+    
+    // Define constants for in/out IDs
+    final int inId = 1;
+    final int inPos = 2;
+    final int outId = 23;
+    
+    public ConnectionModeNGTest() {
+    }
+
+    @BeforeClass
+    public static void setUpClass() throws Exception {
+    }
+
+    @AfterClass
+    public static void tearDownClass() throws Exception {
+    }
+
+    @BeforeMethod
+    public void setUpMethod() throws Exception {
+    }
+
+    @AfterMethod
+    public void tearDownMethod() throws Exception {
+    }
+    
+    /**
+     * Perform reset of all mocks and argument captors to ensure clean test steps.
+     */
+    public void resetMocking() {
+        mockGraphReadMethods = mock(GraphReadMethods.class);
+    }
+
+    /**
+     * Test of values method, of class ConnectionMode.
+     */
+    @Test
+    public void testValues() {
+    }
+
+    /**
+     * Test of valueOf method, of class ConnectionMode.
+     */
+    @Test
+    public void testValueOf() {
+    }
+
+    /**
+     * Test of getConnection method, of class ConnectionMode.
+     */
+    @Test
+    public void testGetConnection() {
+
+        resetMocking();
```

Comment:
if you're needing to reset after each use case, it could be worth separating each one into its own test (and then resetting in setUpMethod). Else in this case (and subsequent tests), there may not be any harm in defining all the whens together and the verifies are then cumulative (the reset can be done inbetween tests as well)

Reply:
I had considered, but then you need to invent a heap of differen ttest case names like "TestGetConnectionWhenSomeFunctionReturnsNull" and "TestGetConnectionWhenSomeFunctionReturnsInvalidId" etc ..... doing it this way test cases can reflect the function under test and have muchcleaner names.

I do the reset as it makes it 'much'clearer to the reader how many instances of a call should occur without having to count back.